samples/desktop/Tiny/SamplesCommon.h

```diff
@@ -0,0 +1,182 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <functional>
+
+namespace igl {
+
+class IDevice;
+
+namespace sample {
+
+enum class Keys {
```

Comment:
I completely understand the motivation to refactor these, but it somewhat goes against the purpose of these demos. The intention behind all the demos in the `samples/` directory is to offer straightforward standalone examples without introducing additional layers or abstractions on top of IGL/GLFW. That's why we haven't factored out any shared code here, and so on. If you're looking for cross-platform wrappers for input and other OS-specific functionalities, there's a set of other examples available at `https://github.com/facebook/igl/tree/main/shell`. Those examples might serve as a better starting point if you're considering introducing abstractions, as suggested in this PR.

Reply:
> I completely understand the motivation to refactor these, but it somewhat goes against the purpose of these demos. The intention behind all the demos in the samples/ directory is to offer straightforward standalone examples without introducing additional layers or abstractions on top of IGL/GLFW. That's why we haven't factored out any shared code here, and so on. 

Well, the purpose of my changes was actually the opposite from what you say, I think. I *don't* want to see any platform code in the examples, the fact that you're using GLFW is just an implementation detail, and it should not be part of them (IMHO of course), it doesn't really help me in the understanding of IGL itself. If I could I'd have no abstraction at all here, but every other graphics library I've used contains some form of "sample harness" in the same vein as the one I've created here for the same reason.

> If you're looking for cross-platform wrappers for input and other OS-specific functionalities, there's a set of other examples available at `https://github.com/facebook/igl/tree/main/shell`. Those examples might serve as a better starting point if you're considering introducing abstractions, as suggested in this PR.

I'll take a look at that other set of examples (also, why are there 2 different sets?), but that makes it sound as I'm introducing the abstractions for the sake of it, but as mentioned, they're the means to an end, not the end in itself 😄.

Reply:
> every other graphics library I've used contains some form of "sample harness" in the same vein as the one I've created here for the same reason.

Then this code here is definitely what you are looking for: https://github.com/facebook/igl/tree/main/shell. If you find issues in there, feel free to create a PR.

> why are there 2 different sets?

As mentioned above, to have both approaches demonstrated - with and without "harness".